modules/asset-manager-storage-aws/src/main/java/org/opencastproject/assetmanager/aws/s3/AwsS3AssetStore.java

```diff
@@ -274,28 +281,33 @@ protected AwsUploadOperationResult uploadObject(File origin, String objectName)
       throw new AssetStoreException("Operation interrupted", e);
     } catch (Exception e) {
       throw new AssetStoreException("Upload failed", e);
-    } finally {
-      try {
-        if (obj != null) {
-          obj.close();
-        }
-      } catch (IOException e) {
-        //Swallow and ignore
-      }
     }
   }
 
   /**
    *
    */
+  @Override
   protected InputStream getObject(AwsAssetMapping map) {
-    S3Object object = s3.getObject(bucketName, map.getObjectKey());
-    return object.getObjectContent();
+    try {
+      // Do not use S3 object stream anymore because the S3 object needs to be closed to release
+      // the http connection so create the stream using the object url (signed).
+      String objectKey = map.getObjectKey();
+      Date expiration = new Date(System.currentTimeMillis() + DOWNLOAD_URL_EXPIRATION_MS);
+      GeneratePresignedUrlRequest generatePresignedUrlRequest = new GeneratePresignedUrlRequest(bucketName, objectKey)
+              .withMethod(HttpMethod.GET).withExpiration(expiration);
+      URL signedUrl = s3.generatePresignedUrl(generatePresignedUrlRequest);
+      logger.debug("Returning pre-signed URL stream for '{}': {}", map, signedUrl);
+      return signedUrl.openStream();
+    } catch (IOException e) {
+      return Prelude.chuck(e);
```

Comment:
Why not raise an `AssetStoreException` here?  `Prelude` is Entwine code, which we're trying to remove :)

Reply:
Good point Greg! I used it because the other archive code uses it, but will remove.